### ISSUE TITLE
combine_vars causes traceback on invalid input

### ISSUE THREAD

**Issue by mdaniel** [ansible/ansible#5557]
The `combine_vars` function is not type-checking its inputs, and is exploding with an `AttributeError`.

```
Traceback (most recent call last):
  File "/usr/local/Cellar/ansible/1.4.4/libexec/bin/ansible-playbook", line 269, in <module>
    sys.exit(main(sys.argv[1:]))
  File "/usr/local/Cellar/ansible/1.4.4/libexec/bin/ansible-playbook", line 209, in main
    pb.run()
  File "/usr/local/Cellar/python/2.7.6/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/ansible/playbook/__init__.py", line 229, in run
    play = Play(self, play_ds, play_basedir)
  File "/usr/local/Cellar/python/2.7.6/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/ansible/playbook/play.py", line 83, in __init__
    ds = self._load_roles(self.roles, ds)
  File "/usr/local/Cellar/python/2.7.6/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/ansible/playbook/play.py", line 327, in _load_roles
    roles = self._build_role_dependencies(roles, [], self.vars)
  File "/usr/local/Cellar/python/2.7.6/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/ansible/playbook/play.py", line 192, in _build_role_dependencies
    role_vars = utils.combine_vars(vars_data, role_vars)
  File "/usr/local/Cellar/python/2.7.6/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/ansible/utils/__init__.py", line 1008, in combine_vars
    return dict(a.items() + b.items())
AttributeError: 'list' object has no attribute 'items'
```

In my attempt to track this down, I inserted

``` python
    assert isinstance(a, dict), 'Expected %r to be a dict but is %s' % (a, type(a))
    assert isinstance(b, dict), 'Expected %r to be a dict but is %s' % (b, type(b))
```

and it gave the list that it was trying to combine, which told me the file that was in error.
I erroneously used

``` yaml

---
- var1: value
- var2: value
```

instead of the correct version without the leading hyphen.

Ansible version: 1.4.4
Python: 2.7.6 built by homebrew

**Comment by mpdehaan** [comment]
Can you share a playbook that reproduces this problem?

There's only a small snippet of a file above and it's definitely not a playbook, so it's unclear how to assemble this together to replicate this message.

**Comment by mdaniel** [comment]
Yes, my apologies:

``` sh
#! /bin/bash
set -e
mkdir -p issue-5557/roles/sample/{tasks,vars}
cat > issue-5557/playbook1.yml<<DATA1
---
- hosts: all
  roles:
    - sample
DATA1
cat > issue-5557/roles/sample/tasks/main.yml<<DATA2
---
- name: echo something
  command: echo hello, world
DATA2
cat > issue-5557/roles/sample/vars/main.yml<<DATA3
---
- bogus: 1
- nope: 2
DATA3
echo 127.0.0.1 | ansible-playbook -vvvv -i /dev/stdin issue-5557/playbook1.yml
```

**Comment by jimi-c** [comment]
This appears to have been resolved at least as far back as 1.5.0:

<pre>
# ansible-playbook -vvvv -i localhost, issue-5557/playbook1.yml
ERROR: vars from '/root/testing/issue-5557/roles/sample/vars/main.yml' are not a dict
</pre>

If you continue to have issues with this, please let us know. Thanks!
